src/AWS.Deploy.Orchestration/Utilities/CloudApplicationNameGenerator.cs

```diff
@@ -68,12 +68,24 @@ public string GenerateValidName(ProjectDefinition target, List<CloudApplication>
             // make sure the recommendation doesn't exist already in existingApplications
             var recommendation = recommendedPrefix;
             var suffix = 1;
+            var recommendationCharArray = recommendation.ToCharArray();
+            var digits = "";
+            for (int i = recommendationCharArray.Length - 1; i >= 0; i--)
+            {
+                if (char.IsDigit(recommendationCharArray[i]))
+                    digits = $"{recommendationCharArray[i]}{digits}";
+                else
+                    break;
+            }
+
+            var parseResult = int.TryParse(digits, out suffix);
+
             while (suffix < 100)
             {
                 if (existingApplications.All(x => x.Name != recommendation) && IsValidName(recommendation))
                     return recommendation;
 
-                recommendation = $"{recommendedPrefix}{suffix++}";
+                recommendation = $"{(parseResult ? recommendation.Substring(0, recommendation.Length - digits.Length) : recommendedPrefix)}{++suffix}";
             }
```

Comment:
some calculations and string parsing be avoided
```csharp
            var recommendationCharArray = recommendation.ToCharArray();
            for (var i = recommendationCharArray.Length - 1; i >= 0; i--)
            {
                if (char.IsDigit(recommendationCharArray[i]))
                    suffix = suffix * 10 + (recommendationCharArray[i] - '0');
                else
                    break;
            }

            var prefix = suffix != 0 ? recommendation[..^suffix.ToString().Length] : recommendedPrefix;
            while (suffix < 100)
            {
                if (existingApplications.All(x => x.Name != recommendation) && IsValidName(recommendation))
                    return recommendation;

                recommendation = $"{prefix}{++suffix}";
            }
```

Reply:
Done